### PR TITLE
starting documentation of test plan

### PR DESCRIPTION
**Pull request by timsaxe**
Signed-off-by: Tim Saxe <saxe@quicklogic.com>

Added a directory sw-based-tests to the test folder.  The purpose of this directory is to contain the documentation that describes the test plan (based on software programs running on emulation or simulation), which are required for RTL freeze and the current status.

### CODE REVIEW

**Comment by zarubaf** [comment]
That's great, wouldn't it be really cool if we can just plug it into the existing documentation and let it be rendered as part of it? Otherwise, we can also merge now and maybe unify later ;-)

**Comment by timsaxe** [comment]
We could do that.  I wasn’t sure if we wanted the test plan as part of the user doc, but it does make it more convenient.  The follow-on question is do we move the sources to the docs folder or leave them in tests.  I put them in tests because I thought that is where someone who was interested in testing would naturally look.

**Comment by zarubaf** [comment]
Yeah, I guess both places make sense, for the sake of accessing it, it probably makes sense to have everything in a single place. No hard opinion though :-)

**Comment by timsaxe** [comment]
I combined the docs into a single doc with three chapters.  Sphinx appears not to like ../ so I moved the stuff from test into docs.

File: docs/auto-docs/apb_gpio_reg_defs.md [hunk 1]
```diff
@@ -1,144 +0,0 @@
-# APB_GPIO
```
Comment (zarubaf) [inline docs/auto-docs/apb_gpio_reg_defs.md:1]: I am wondering why this is being deleted here, does that make sense to you?

**Review by zarubaf — APPROVED** [review]

**Comment by zarubaf** [comment]
I think that has been merged as part of #131. If not please feel free to re-open.

**Pull request closed without merging**